data-prepper-api/src/test/java/com/amazon/dataprepper/model/configuration/SinkModelTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.model.configuration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasKey;
+
+class SinkModelTest {
+    private ObjectMapper objectMapper;
+
+    @BeforeEach
+    void setup() {
+        objectMapper = new ObjectMapper(YAMLFactory.builder()
+                .disable(YAMLGenerator.Feature.WRITE_DOC_START_MARKER)
+                .enable(YAMLGenerator.Feature.USE_PLATFORM_LINE_BREAKS)
+                .build());
+    }
+
+    @Test
+    void deserialize_with_Sink() throws IOException {
+        final InputStream inputStream = this.getClass().getResourceAsStream("sink_plugin.yaml");
+
+        final SinkModel sinkModel = objectMapper.readValue(inputStream, SinkModel.class);
+
+        assertThat(sinkModel, notNullValue());
+        assertThat(sinkModel.getPluginName(), equalTo("customSinkPlugin"));
+        assertThat(sinkModel.getPluginSettings(), notNullValue());
+        assertThat(sinkModel.getPluginSettings().size(), equalTo(2));
+        assertThat(sinkModel.getPluginSettings(), hasKey("key1"));
+        assertThat(sinkModel.getPluginSettings(), hasKey("key2"));
+        assertThat(sinkModel.getRoutes(), notNullValue());
+        assertThat(sinkModel.getRoutes().size(), equalTo(2));
+        assertThat(sinkModel.getRoutes(), hasItem("routeA"));
+        assertThat(sinkModel.getRoutes(), hasItem("routeB"));
+    }
+
+    @Test
+    void serialize_into_known_SinkModel() throws IOException {
+        final Map<String, Object> pluginSettings = new LinkedHashMap<>();
+        pluginSettings.put("key1", "value1");
+        pluginSettings.put("key2", "value2");
+        final SinkModel sinkModel = new SinkModel("customSinkPlugin", Arrays.asList("routeA", "routeB"), pluginSettings);
+
+        final String actualJson = objectMapper.writeValueAsString(sinkModel);
+
+        final String expectedJson = createStringFromInputStream(this.getClass().getResourceAsStream("sink_plugin.yaml"))
+                .replace("---\n", "");
```

Comment:
Consider covering the case where createStringFromInputStream produces multiple YAML documents by
adding "---\n" to actualJson or assert the pattern "---\n" occurs only x times before calling replace.